system-addon/common/Reducers.jsm

```diff
@@ -334,6 +334,8 @@ function Snippets(prevState = INITIAL_STATE.Snippets, action) {
   switch (action.type) {
     case at.SNIPPETS_DATA:
       return Object.assign({}, prevState, {initialized: true}, action.data);
+    case at.SNIPPET_BLOCKED:
```

Comment:
The original implementation was kind of a stop-gap measure, but I think it might be a better  if instead of passing the entire block list around every time, we just send a single ID and append it here.